src/modules/collection/sagas.ts

```diff
@@ -545,13 +548,26 @@ export function* collectionSaga(builder: BuilderAPI, catalyst: CatalystClient) {
 
       // 1. Open modal
 
-      // 2. Get the approval data from the server (using the builder client)
+      // 2. Get the approval data from the server
+      // TODO: Use the builder client. Tracked here: https://github.com/decentraland/builder/issues/1855
+      const { cheque, content_hashes }: ItemApprovalData = yield call([builder, 'fetchApprovalData'], collection.id)
 
       // 3. Compute the merkle tree root
+      const tree = generateTree(content_hashes)
 
       // Open the ApprovalFlowModal with the items to be approved
 
       // 4. Make the transaction to the contract (update of the merkle tree root with the signature and its parameters)
+      const maticChainId: ChainId = yield call(getChainIdByNetwork, Network.MATIC)
+      const thirdPartyId = extractThirdPartyId(collection.urn)
+      const root = tree.merkleRoot
+      const { r, s, v } = ethers.utils.splitSignature(cheque.signature)
+      const thirdPartyContract: ContractData = yield call(getContract, ContractName.ThirdPartyRegistry, maticChainId)
+      const txHash: string = yield call(sendTransaction, thirdPartyContract, contract =>
```

Comment:
I think that the `reviewThirdPartyWithRoot`'s 3rd parameter is an array of `ConsumeSlotsParam` so we might be missing the `qty` and `salt` for our cheque.

Would it be something like the following? (I haven't test it)
```
contract.reviewThirdPartyWithRoot(thirdPartyId, root, [{ signR: r, signS: s, signV: v, qty, salt  }])
```


Also related to this, should we use what @LautaroPetaccio did so the `sendTransaction` is easier to test? 
```
await sendTransaction(contract, 'reviewThirdPartyWithRoot', thirdPartyId, root, [cheque]`
```

Reply:
oh shoot! right on both accounts. I'll give a it a go and test it